mnylc/islandora_multi_importer
# Mnylc issue93

[**Ticket 93**](https://github.com/mnylc/islandora_multi_importer/issues/93)

# What does this Pull Request do?

* When fetching remote files, the root of the public:// files directory can become filled with very large files, necessitating massive storage where it may not be appropriate.
* Temporary files and metadata for batch ingests should not be stored in a location that is freely open to the public.

# What's new?
* Makes the temporary file storage location configurable.
* Puts temporary file storage under the private:// directory, rather than public://

# How should this be tested?

* Prepare and process an IMI ingest which uses Amazon S3 for file source. The uploaded csv file should be found in private://islandora-multi-importer/csv. The ingest should be successful, and the fetched files should be saved into private://islandora-multi-importer.
* Go to the imi settings page (/admin/islandora/tools/multi_importer) and change the "Temporary files storage directory". When running a new ingest, the directory should get created, and the csv and downloaded files should be saved to folders within this directory.
* When deleting an ingest batch, the temporary files should be removed from wherever they were stored, even if the storage location configuration changed.

# Additional Notes:
* Does this change require documentation to be updated? 
Yes, the documentation for the IMI settings page should be updated. 
* Does this change add any new dependencies? 
No
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? 
No.
* Could this change impact execution of existing code?
I don't think so.

# Interested parties
@DiegoPino
